packages/backend/src/managers/playground/McpServerManager.ts

```diff
@@ -0,0 +1,75 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { Disposable } from '@podman-desktop/api';
+import fs from 'node:fs';
+import path from 'node:path';
+
+const MCP_SETTINGS = 'mcp-settings.json';
+
+export enum McpServerType {
+  STDIO = 'stdio',
+  SSE = 'sse',
+}
+
+export interface McpServer {
+  name: string;
+  enabled: boolean;
+  type: McpServerType;
+  command: string;
+  args: string[];
+  url: string;
+  headers: Record<string, string>;
+}
+
+// TODO: Consider using JsonWatcher in case the file is going to be updated externally
+export class McpServerManager implements Disposable {
+  private readonly settingsFile: string;
+
+  constructor(private appUserDirectory: string) {
+    this.settingsFile = path.join(this.appUserDirectory, MCP_SETTINGS);
+  }
+
+  async load(): Promise<McpServer[]> {
```

Comment:
Shouldn't we keep them in memory? The current logic does not save them

Reply:
Still not sure about this one: https://github.com/containers/podman-desktop-extension-ai-lab/pull/2916#discussion_r2055923268

Reply:
Just to be more clear. This is the initial implementation of the manager, so that I can actually bind it to the playground manager and enable tool calling.
Depending on how the tools are managed we should keep this in memory or expand the management capabilities.